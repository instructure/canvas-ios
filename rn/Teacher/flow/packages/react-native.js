//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
