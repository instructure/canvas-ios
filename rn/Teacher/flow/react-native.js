//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
