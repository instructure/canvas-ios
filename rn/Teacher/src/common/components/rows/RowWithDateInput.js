//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import React, { PureComponent } from 'react'
import {
  View,
  TouchableHighlight,
  Image,
} from 'react-native'
import { Text } from '../../text'
import i18n from 'format-message'
import { colors, createStyleSheet, vars } from '../../stylesheet'
import Images from '../../../images'
import { extractDateFromString } from '../../../utils/dateUtils'

type DateRowProps = {
  title: string,
  date: ?string,
  showRemoveButton?: boolean,
  onPress?: Function,
  onRemoveDatePress?: Function,
  testID?: string,
  dateTestID?: string,
  removeButtonTestID?: string,
  selected?: boolean,
}

export default class RowWithDateInput extends PureComponent<DateRowProps, any> {
  render () {
    let detailTextStyle = this.props.selected ? { color: colors.primary } : styles.detailText
    let paddingRightWhenEmpty = this.props.showRemoveButton ? 0 : vars.padding

    let date = extractDateFromString(this.props.date)
    date = date ? i18n("{ date, date, 'MMM d' } { date, time, short }", { date }) : i18n('--')
    return (
      <View style={[ styles.row, styles.detailsRowContainer ]} >
        <TouchableHighlight style={{ flex: 1 }} accessibilityTraits={['button']} onPress={this.props.onPress} testID={this.props.testID}>
          <View style={[styles.rowContainer, { paddingRight: paddingRightWhenEmpty }]}>
            <View style={styles.titlesContainer}>
              <Text style={styles.titleText}>{this.props.title}</Text>
            </View>
            <View style={styles.detailsRowContainer}>
              <Text style={detailTextStyle} testID={this.props.dateTestID}>{date}</Text>
            </View>
          </View>
        </TouchableHighlight>
        { this.props.showRemoveButton &&
          <View style={styles.deleteDateTypeButton}>
            <TouchableHighlight
              underlayColor='transparent'
              onPress={this.props.onRemoveDatePress}
              accessible={true}
              accessibilityLabel={i18n('Remove date')}
              accessibilityTraits='button'
              testID={this.props.removeButtonTestID}
              containerStyle={styles.deleteDateTypeButton}
            >
              <Image source={Images.clear} />
            </TouchableHighlight>
          </View>
        }
      </View>
    )
  }
}

const styles = createStyleSheet((colors, vars) => ({
  row: {
    paddingVertical: vars.padding / 2,
    borderBottomWidth: vars.hairlineWidth,
    borderBottomColor: colors.borderMedium,
    backgroundColor: colors.backgroundLightest,
    minHeight: 54,
  },
  rowContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: colors.backgroundLightest,
    paddingLeft: vars.padding,
  },
  detailsRowContainer: {
    justifyContent: 'flex-end',
    flexDirection: 'row',
    alignItems: 'center',
    minWidth: 90,
  },
  titlesContainer: {
    paddingRight: 32,
  },
  titleText: {
    fontWeight: '600',
  },
  detailText: {
  },
  deleteDateTypeButton: {
    paddingTop: 8,
    paddingBottom: 8,
    paddingLeft: 8,
    paddingRight: vars.padding,
    alignItems: 'center',
    justifyContent: 'center',
  },
}))
