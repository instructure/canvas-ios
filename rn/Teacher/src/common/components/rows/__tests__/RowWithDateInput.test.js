//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import 'react-native'
import React from 'react'
import RowWithDateInput from '../RowWithDateInput'

import renderer from 'react-test-renderer'

test('Render row with date', () => {
  let aRow = renderer.create(
    <RowWithDateInput
      title='Row with a date in it!'
      date='this should be a date'
    />
  )
  expect(aRow.toJSON()).toMatchSnapshot()
})

test('Render row with date and the clear button', () => {
  let aRow = renderer.create(
    <RowWithDateInput
      title='Row with a date in it!'
      date='this should be a date'
      showRemoveButton={true}
    />
  )
  expect(aRow.toJSON()).toMatchSnapshot()
})
