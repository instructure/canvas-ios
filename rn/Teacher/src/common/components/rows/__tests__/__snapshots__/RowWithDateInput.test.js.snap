// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render row with date 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#C7CDD1",
        "borderBottomWidth": 0.5,
        "minHeight": 54,
        "paddingVertical": 8,
      },
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "minWidth": 90,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
          },
          Object {
            "paddingRight": 16,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "paddingRight": 32,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "fontWeight": "600",
              },
            ]
          }
        >
          Row with a date in it!
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "minWidth": 90,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {},
            ]
          }
        >
          --
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Render row with date and the clear button 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#C7CDD1",
        "borderBottomWidth": 0.5,
        "minHeight": 54,
        "paddingVertical": 8,
      },
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "minWidth": 90,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
          },
          Object {
            "paddingRight": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "paddingRight": 32,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "fontWeight": "600",
              },
            ]
          }
        >
          Row with a date in it!
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "minWidth": 90,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {},
            ]
          }
        >
          --
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 16,
        "paddingTop": 8,
      }
    }
  >
    <View
      accessibilityLabel="Remove date"
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/Clear.png",
          }
        }
      />
    </View>
  </View>
</View>
`;
